cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsMetadataCollector.java

```diff
@@ -252,4 +276,50 @@ private void addKnownInstance(CloudInstance cloudInstance, Multimap<String, Inst
     public InstanceStoreMetadata collectInstanceStorageCount(AuthenticatedContext ac, List<String> instanceTypes) {
         return awsPlatformResources.collectInstanceStorageCount(ac, instanceTypes);
     }
+
+    /**
+     * Parses the stack resource summary for the cloudformation stack and pulls out all listener and target group ARNs
+     * associated with a particular load balancer, and creates AwsTargetGroupMetadata objects using the provided
+     * resources summaries.
+     * @param scheme The scheme of the load balancer being processed, either INTERNAL or INTERNET-FACING
+     * @param summaries The list of resource summaries from cloud formation that contain both the logical resource id
+     *                  (the name), and the physical resource id (ARN).
+     * @return A list of metadata pulled from the resource summaries.
+     */
+    private List<AwsTargetGroupMetadata> parseTargetGroupCloudConfig(AwsLoadBalancerScheme scheme, List<StackResourceSummary> summaries) {
```

Comment:
Can we create a separate (util) class to cover this functionality instead of extending the metadata collector?
I think it would result in cleaner code and easier to be unit tested.

Reply:
Moved this to a new `AwsLoadBalancerMetadataCollectorClass`.